### PR TITLE
feat(usecase): implement usecase backend logic

### PR DESCRIPTION
**Pull request by nidhigarg-bmw**
## Description

Add UseCase business logic functionality

## Why

To get the list of usecase with credentials and can update them

## Issue

CPLP-2851
CPLP-2856

## Checklist

Please delete options that are not relevant.

- [X] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [X] I have performed a self-review of my own code
- [X] I have successfully tested my changes locally

### CODE REVIEW

**Review by evegufy — CHANGES_REQUESTED** [review]
Contributing guidelines aren't followed: PR title and description need to be adjusted and improved. Also, links to Jira need to be removed in the PR description as this references to a closed system: if at all, mention the Jira issue ID

**Review by evegufy — CHANGES_REQUESTED** [review]
PR title still needs to be adjusted according contributing guidelines

**Review by lavanya-bmw — CHANGES_REQUESTED** [review]
Please remove console logs

File: src/features/usecase/usecaseApiSlice.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
+/********************************************************************************
+ * Copyright (c) 2021, 2023 BMW Group AG
```
Comment (manojava-gk) [inline src/features/usecase/usecaseApiSlice.ts:2]: remove company specific header

**Review by manojava-gk — COMMENTED** [review]
Small finding

**Review by lavanya-bmw — APPROVED** [review]
lgtm

**Review by oyo — DISMISSED** [review]

**Review by evegufy — APPROVED** [review]

**Review by evegufy — CHANGES_REQUESTED** [review]
please update the dependencies file:
https://github.com/eclipse-tractusx/portal-frontend/actions/runs/5442501820/jobs/9897815989?pr=110
download artifact and commit:
https://github.com/eclipse-tractusx/portal-frontend/actions/runs/5442501820?pr=110

**Review by evegufy — APPROVED** [review]

**Pull request merged**